File: RaftSwiftGRPC/Sources/RaftCore/Utils/Types/Snapshot.swift
/// A snapshot of the entire state machine
struct Snapshot {
    /// Index of last log entry included in the snapshot
    var lastIncludedIndex: Int = 0

    /// Term of last log entry included in the snapshot
    var lastIncludedTerm: Int = 0

    /// State machine state
    var stateMachine: [String: String] = [:]
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/core/utils/types/Snapshot.kt
package raft.core.utils.types

import kotlinx.serialization.Serializable

/**
 * A snapshot of the state machine
 */
@Serializable
data class Snapshot(
    /**
     * Index of last log entry included in the snapshot
     */
    val lastIncludedIndex: Int = 0,
    /**
     * Term of last log entry included in the snapshot
     */
    val lastIncludedTerm: Int = 0,
    /**
     * State machine state
     */
    val stateMachine: MutableMap<String, String> = mutableMapOf()
)